core/src/main/java/org/elasticsearch/discovery/DiscoverySettings.java

```diff
@@ -88,47 +89,25 @@ public ClusterBlock getNoMasterBlock() {
         return noMasterBlock;
     }
 
-    public boolean getPublishDiff() { return publishDiff;}
+    private void setNoMasterBlock(ClusterBlock noMasterBlock) {
+        this.noMasterBlock = noMasterBlock;
+    }
 
-    private class ApplySettings implements NodeSettingsService.Listener {
-        @Override
-        public void onRefreshSettings(Settings settings) {
-            TimeValue newPublishTimeout = settings.getAsTime(PUBLISH_TIMEOUT, null);
-            if (newPublishTimeout != null) {
-                if (newPublishTimeout.millis() != publishTimeout.millis()) {
-                    logger.info("updating [{}] from [{}] to [{}]", PUBLISH_TIMEOUT, publishTimeout, newPublishTimeout);
-                    publishTimeout = newPublishTimeout;
-                    if (settings.getAsTime(COMMIT_TIMEOUT, null) == null && commitTimeout.millis() > publishTimeout.millis()) {
```

Comment:
I'm ++ on thrashing the auto magic wiring of defaults. I do think we need to validate the publish timeout is equal or higher than the commit timeout, o.w. it doesn't make sense. It seems we now have the intra to do it simply?